controllers/add_pods.go

```diff
@@ -245,6 +250,32 @@ func createInstance(r *FoundationDBClusterReconciler, context ctx.Context, clust
 			err = r.Create(context, service)
 			return false, err
 		}
+
+		// If we create a new Pod and reuse a service we should ensure that it matches.
+		if processClass == fdbtypes.ProcessClassStorage {
+			// Ensure that the service contains all required ports.
+			// Since a process can listen on non-TLS and TLS we multiply by 2.
+			// Since we are creating a new Pod we can use the cluster.GetStorageServersPerPod()
+			// if this value would be changed during the creation it should be reconciled again.
+			if len(services.Items[0].Spec.Ports) != (cluster.GetStorageServersPerPod() * 2) {
```

Comment:
This seems like an odd place to update service config. We've got another issue on allowing customization of the service spec, so maybe we should think about where we want to update service config in that context, and whether updating the ports can work in the same place.

Reply:
The idea here was that If a Pod was replaced with the same `instance_id` and there already exists an service with the given `instance_id` we need to ensure that all required ports are exposed.

Reply:
I think for the current stage this is fine but probably once we got a more complex/richer reconciliation for services we want to move it there.